### PR TITLE
Added the release 1.4.1-ES properties

### PR DESCRIPTION
**Pull request by kaifk468**

### CODE REVIEW

File: esignet-default.properties [hunk 1]
```diff
@@ -134,7 +161,7 @@ mosip.esignet.authenticator.ida.otp-channels=email,phone
 mosip.esignet.ida.vci-user-info-cache=userinfo
 mosip.esignet.ida.vci-exchange-id=mosip.identity.vciexchange
 mosip.esignet.ida.vci-exchange-version=1.0
-mosip.esignet.ida.vci-exchange-url=${mosip.ida.auth.url}/idauthentication/v1/vci-exchange/delegated/${mosip.esignet.misp.license.key}/
+mosip.esignet.ida.vci-exchange-url=https://${mosip.api.internal.host}/idauthentication/v1/vci-exchange/delegated/${mosip.esignet.misp.license.key}/
```
Comment (ghost) [inline esignet-default.properties]: change this  to previous value ${mosip.ida.auth.url}

File: esignet-default.properties [hunk 2]
```diff
@@ -199,25 +226,22 @@ mosip.esignet.cache.store.individual-id=true
 mosip.esignet.cache.security.secretkey.reference-id=TRANSACTION_CACHE
 mosip.esignet.cache.security.algorithm-name=AES/ECB/PKCS5Padding
 
-mosip.esignet.ida.vci-user-info-cache=userinfo
+mosip.esignet.cache.names=clientdetails,preauth,authenticated,authcodegenerated,userinfo,linkcodegenerated,linked,linkedcode,linkedauth,consented,authtokens,bindingtransaction,vcissuance,apiRateLimit,blocked
```
Comment (ghost) [inline esignet-default.properties]: apiRateLimit -> apiratelimit

File: esignet-default.properties [hunk 3]
```diff
@@ -199,25 +226,22 @@ mosip.esignet.cache.store.individual-id=true
...
 
-mosip.esignet.cache.names=clientdetails,preauth,authenticated,authcodegenerated,userinfo,linkcodegenerated,linked,linkedcode,linkedauth,consented,authtokens,bindingtransaction,vcissuance
-
-spring.cache.type=redis
```
Comment (ghost) [inline esignet-default.properties:206]: FOr performance testing we should connect to redis

File: esignet-default.properties [hunk 4]
```diff
@@ -199,25 +226,22 @@ mosip.esignet.cache.store.individual-id=true
...
-#spring.cache.type=simple
+spring.cache.type=simple
 mosip.esignet.cache.key.hash.algorithm=SHA3-256
 
-# Cache size setup is applicable only for 'simple' cache type. 
+# Cache size setup is applicable only for 'simple' cache type.
 # Cache size configuration will not be considered with 'Redis' cache type
 mosip.esignet.cache.size={'clientdetails' : 200, \
-'preauth': 2000, \
+'preauth': 200, \
```
Comment (ghost) [inline esignet-default.properties]: Change the value to 2000 -> because for performance testing we will need higher preauth transaction time

File: esignet-default.properties [hunk 5]
```diff
@@ -228,13 +252,15 @@ mosip.esignet.cache.size={'clientdetails' : 200, \
 'consented' :200, \
 'authtokens': 2, \
 'bindingtransaction': 200, \
-'vcissuance' : 200 }
+'vcissuance' : 200, \
+'apiRateLimit' : 500, \
```
Comment (ghost) [inline esignet-default.properties]: apiRateLimit -> apiratelimit

File: esignet-default.properties [hunk 6]
```diff
@@ -243,14 +269,16 @@ mosip.esignet.cache.expire-in-seconds={'clientdetails' : 86400, \
 'consented': 60, \
 'authtokens': 86400, \
 'bindingtransaction': 600, \
-'vcissuance': ${mosip.esignet.access-token-expire-seconds}  }
+'vcissuance': ${mosip.esignet.access-token-expire-seconds}, \
+'apiRateLimit' : 180, \
```
Comment (ghost) [inline esignet-default.properties]: apiRateLimit -> apiratelimit

File: esignet-default.properties [hunk 7]
```diff
@@ -228,13 +252,15 @@ mosip.esignet.cache.size={'clientdetails' : 200, \
...
 'bindingtransaction': 200, \
-'vcissuance' : 200 }
+'vcissuance' : 200, \
+'apiRateLimit' : 500, \
+'blocked': 500 }
 
 # Cache expire in seconds is applicable for both 'simple' and 'Redis' cache type
 mosip.esignet.cache.expire-in-seconds={'clientdetails' : 86400, \
-'preauth': 86400,\
+'preauth': 300,\
```
Comment (ghost) [inline esignet-default.properties]: Change the value to 86400 -> because for performance testing we will need higher preauth transaction time

File: esignet-default.properties [hunk 8]
```diff
@@ -228,13 +252,15 @@ mosip.esignet.cache.size={'clientdetails' : 200, \
...
 'authenticated': ${mosip.esignet.authentication-expire-in-secs}, \
-'authcodegenerated': 86400, \
+'authcodegenerated': 60, \
```
Comment (ghost) [inline esignet-default.properties]: Change the value to 86400 -> because for performance testing we will need higher authcodegenerated transaction time

File: esignet-default.properties [hunk 9]
```diff
@@ -274,26 +302,26 @@ mosip.esignet.discovery.key-values={'issuer': '${mosip.esignet.domain.url}' ,\
...
+  \ 'acr_values_supported' : {'mosip:idp:acr:static-code', 'mosip:idp:acr:generated-code', 'mosip:idp:acr:linked-wallet', 'mosip:idp:acr:biometrics', 'mosip:idp:acr:knowledge'},\
+  \ 'request_parameter_supported' : false, \
   \ 'claims_locales_supported' : {'en'}, \
-  \ 'display_values_supported' : ${mosip.esignet.supported.ui.displays}, \
   \ 'ui_locales_supported' : {'en'} }
 
 ##----------------------------------------- Database properties --------------------------------------------------------
 
-mosip.esignet.database.hostname=172.31.13.139
+mosip.esignet.database.hostname=postgres-postgresql.postgres
```
Comment (ghost) [inline esignet-default.properties]: Because its performance env, we are using external DB, change the value back to 172.31.13.139

File: esignet-default.properties [hunk 10]
```diff
@@ -337,11 +365,11 @@ mosip.kernel.certificate.sign.algorithm=SHA256withRSA
 #mosip.kernel.keymanager.hsm.keystore-pass=${softhsm.idp.pin}
 
 #Type of keystore, Supported Types: PKCS11, PKCS12, Offline, JCE
-mosip.kernel.keymanager.hsm.keystore-type=PKCS12
+mosip.kernel.keymanager.hsm.keystore-type=PKCS11
```
Comment (ghost) [inline esignet-default.properties]: For perfoamnce, we should test with PKCS12, kindly revert this change

File: esignet-default.properties [hunk 11]
```diff
@@ -337,11 +365,11 @@ mosip.kernel.certificate.sign.algorithm=SHA256withRSA
...
 # For PKCS11 provide Path of config file.
 # For PKCS12 keystore type provide the p12/pfx file path. P12 file will be created internally so provide only file path & file name.
 # For Offline & JCE property can be left blank, specified value will be ignored.
-mosip.kernel.keymanager.hsm.config-path=/home/mosip/config/keys.p12
+mosip.kernel.keymanager.hsm.config-path=/config/softhsm-application.conf
```
Comment (ghost) [inline esignet-default.properties]: For perfoamnce, we should test with PKCS12, kindly revert this change

File: esignet-default.properties (hunk 1 above)
Comment (kaifk468) [reply esignet-default.properties]: done

File: esignet-default.properties (hunk 2 above)
Comment (kaifk468) [reply esignet-default.properties]: done

File: esignet-default.properties (hunk 4 above)
Comment (kaifk468) [reply esignet-default.properties]: done

File: esignet-default.properties (hunk 5 above)
Comment (kaifk468) [reply esignet-default.properties]: done

File: esignet-default.properties (hunk 6 above)
Comment (kaifk468) [reply esignet-default.properties]: done

File: esignet-default.properties (hunk 7 above)
Comment (kaifk468) [reply esignet-default.properties]: reverted

File: esignet-default.properties (hunk 8 above)
Comment (kaifk468) [reply esignet-default.properties]: reverted

File: esignet-default.properties (hunk 9 above)
Comment (kaifk468) [reply esignet-default.properties]: reverted

File: esignet-default.properties (hunk 10 above)
Comment (kaifk468) [reply esignet-default.properties]: reverted

File: esignet-default.properties (hunk 3 above)
Comment (kaifk468) [reply esignet-default.properties:206]: reverted to redis

File: esignet-default.properties [hunk 12]
```diff
@@ -28,31 +28,59 @@
...
   '${server.servlet.path}/authorization/v3/authenticate', \
   '${server.servlet.path}/authorization/auth-code'}
 
-mosip.esignet.captcha.required=send-otp,pwd,bio,pin
-
-mosip.esignet.captcha.required.auth-factors={'PWD'}
-mosip.esignet.captcha.required.auth-factors={}
+#This property is used for captcha validation and allowed values are send-otp, pwd and kba.
+#captcha validation is enabled for send-otp, pwd and kba.
+mosip.esignet.captcha.required=send-otp,pwd
```
Comment (ghost) [inline esignet-default.properties]: We should disable captcha, as its performance testing

File: esignet-default.properties [hunk 13]
```diff
@@ -28,31 +28,59 @@
...
-mosip.esignet.access-token-expire-seconds=86400
+mosip.esignet.id-token-expire-seconds=3600
+mosip.esignet.access-token-expire-seconds=3600
 # By default, only 2 link codes can be active, and the time period it can be active is defined here, default value is 1 minute
 mosip.esignet.link-code-expire-in-secs=60
 # Number of link code allowed to be generated in a transaction, the default value is 10
 mosip.esignet.generate-link-code.limit-per-transaction=10
 # Time to complete consent after successful authentication, the default value is 120
-mosip.esignet.authentication-expire-in-secs=86400
+mosip.esignet.authentication-expire-in-secs=120
```
Comment (ghost) [inline esignet-default.properties]: Let's revert all the timeout changes.

File: esignet-default.properties [hunk 14]
```diff
@@ -97,16 +125,15 @@ mosip.esignet.kafka.linked-auth-code.topic=esignet-consented
 
 ## ------------------------------------------- Integrations ------------------------------------------------------------
 
-mosip.esignet.integration.scan-base-package=io.mosip.esignet.mock.integration,io.mosip.authentication.esignet.integration
+mosip.esignet.integration.scan-base-package=io.mosip.authentication.esignet.integration,io.mosip.esignet.mock.integration
 mosip.esignet.integration.binding-validator=BindingValidatorServiceImpl
 mosip.esignet.integration.authenticator=IdaAuthenticatorImpl
 mosip.esignet.integration.key-binder=IdaKeyBinderImpl
-mosip.esignet.integration.audit-plugin=LoggerAuditService
+mosip.esignet.integration.audit-plugin=IdaAuditPluginImpl
```
Comment (ghost) [inline esignet-default.properties]: For now for testing lets switch to LoggerAuditService

File: esignet-default.properties [hunk 15]
```diff
@@ -28,31 +28,59 @@
...
 mosip.esignet.misp.license.key=${mosip.esignet.misp.key}
-mosip.esignet.amr-acr-mapping-file-url=${config.server.file.storage.uri}amr-acr-mapping.json
+mosip.esignet.amr-acr-mapping-file-url=${spring_config_url_env}/*/${active_profile_env}/${spring_config_label_env}/amr-acr-mapping.json
 mosip.esignet.auth-txn-id-length=10
 mosip.esignet.supported-id-regex=\\S*
 # Generated ID and access tokens 'exp' depends on the below properties, default value is 1-hour
-mosip.esignet.id-token-expire-seconds=86400
-mosip.esignet.access-token-expire-seconds=86400
+mosip.esignet.id-token-expire-seconds=3600
+mosip.esignet.access-token-expire-seconds=3600
```
Comment (ghost) [inline esignet-default.properties]: Let's revert all the timeout changes.

File: esignet-default.properties (hunk 12 above)
Comment (kaifk468) [reply esignet-default.properties]: disabled

File: esignet-default.properties (hunk 13 above)
Comment (kaifk468) [reply esignet-default.properties]: reverted all the timeout changes

File: esignet-default.properties (hunk 14 above)
Comment (kaifk468) [reply esignet-default.properties]: switched

File: esignet-default.properties (hunk 15 above)
Comment (kaifk468) [reply esignet-default.properties]: reverted
